pydicom/fileset.py

```diff
@@ -1508,34 +1508,40 @@ def find_values(
 
         Returns
         -------
-        list of object
-            A list of value(s) for the element available in the instances.
+        list of object(s), or dict of lists of object(s)
+            If single element was queried: A list of value(s) for the element
+                available in the instances.
+            If list of elements was queried: A dict of element value pairs with
+                lists of value(s) for the elements available in the instances.
         """
-        has_element = False
-        results = []
+        element_list = elements if isinstance(elements, list) else [elements]
+        has_element = {element: False for element in element_list}
+        results: Dict = {element: [] for element in element_list}
         iter_instances = instances or iter(self)
         instance: Union[Dataset, FileInstance]
         for instance in iter_instances:
             if load:
                 instance = instance.load()
-
-            if element not in instance:
-                continue
-
-            has_element = True
-            val = instance[element].value
-            # Not very efficient, but we can't use set
-            if val not in results:
-                results.append(val)
-
-        if not load and not has_element:
-            warnings.warn(
-                "None of the records in the DICOMDIR dataset contain "
-                "the query element, consider using the 'load' parameter "
-                "to expand the search to the corresponding SOP instances"
-            )
-
-        return results
+            for element in element_list:
+                if element not in instance:
+                    continue
+
+                has_element[element] = True
+                val = instance[element].value
+                # Not very efficient, but we can't use set
+                if val not in results[element]:
+                    results[element].append(val)
+        for element, v in has_element.items():
+            if not load and not v:
+                warnings.warn(
+                    "None of the records in the DICOMDIR dataset contain "
+                    f"{element}, consider using the 'load' parameter "
+                    "to expand the search to the corresponding SOP instances"
+                )
```

Comment:
Would be better if you created a list of the elements so theres a single warning